Calling load_table().scan().to_arrow() emits error on empty "names" field in name mapping

### Apache Iceberg version

main (development)

### Please describe the bug 🐞

On one of my iceberg tables, when I load a table and scan it, during the parsing of the name mapping in the table properties, pydantic issues the following `ValidationError`:

```
    def parse_mapping_from_json(mapping: str) -> NameMapping:
>       return NameMapping.model_validate_json(mapping)
E       pydantic_core._pydantic_core.ValidationError: 1 validation error for NameMapping
E       9.names
E         Value error, At least one mapped name must be provided for the field [type=value_error, input_value=[], input_type=list]
E           For further information visit https://errors.pydantic.dev/2.8/v/value_error
```

This seems to be a result of the code in `table/name_mapping.py` in the method `check_at_least_one`, which (if I understand correctly) checks that all fields in the name mapping have at least one `name`. However, if I'm reading the [Iceberg spec](https://iceberg.apache.org/spec/#column-projection) correctly, it states that:

![image](https://github.com/user-attachments/assets/849de6b3-ccfd-4845-9d6a-58b81883784e)

I'm not 100% sure what scenario lead to this but I can say that the name mapping we have indeed has a field with id 10 that has an empty list of names. This field existed at one point in the schema but it seems like it was removed. In any case, it doesn't seem like requiring that the list of names contain at least one value is in line with the spec (and it seems that situations where this isn't the case do happen). 

Note that the said iceberg table was never created, written to or modified using `pyiceberg` (only using spark and trino). `pyiceberg` is only used to read.

--- a/pyiceberg/table/name_mapping.py
+++ b/pyiceberg/table/name_mapping.py
@@ -45,18 +45,6 @@ class MappedField(IcebergBaseModel):
     def convert_null_to_empty_List(cls, v: Any) -> Any:
         return v or []
 
-    @field_validator("names", mode="after")
-    @classmethod
-    def check_at_least_one(cls, v: List[str]) -> Any:
-        """
-        Conlist constraint does not seem to be validating the class on instantiation.
-
-        Adding a custom validator to enforce min_length=1 constraint.
-        """
-        if len(v) < 1:
-            raise ValueError("At least one mapped name must be provided for the field")
-        return v
-
     @model_serializer
     def ser_model(self) -> Dict[str, Any]:
         """Set custom serializer to leave out the field when it is empty."""

--- a/pyiceberg/table/name_mapping.py
+++ b/pyiceberg/table/name_mapping.py
@@ -37,7 +37,7 @@ from pyiceberg.types import ListType, MapType, NestedField, PrimitiveType, Struc
 
 class MappedField(IcebergBaseModel):
     field_id: int = Field(alias="field-id")
-    names: List[str] = conlist(str, min_length=1)
+    names: List[str] = conlist(str)
     fields: List[MappedField] = Field(default_factory=list)
 
     @field_validator("fields", mode="before")

--- a/tests/table/test_name_mapping.py
+++ b/tests/table/test_name_mapping.py
@@ -90,6 +90,22 @@ def test_json_mapped_field_deserialization() -> None:
     """
     assert MappedField(field_id=1, names=["id", "record_id"]) == MappedField.model_validate_json(mapped_field_with_null_fields)
 
+def test_json_mapped_field_no_names_deserialization() -> None:
+    mapped_field = """{
+        "field-id": 1,
+        "names": []
+    }
+    """
+    assert MappedField(field_id=1, names=[]) == MappedField.model_validate_json(mapped_field)
+
+    mapped_field_with_null_fields = """{
+        "field-id": 1,
+        "names": [],
+        "fields": null
+    }
+    """
+    assert MappedField(field_id=1, names=[]) == MappedField.model_validate_json(mapped_field_with_null_fields)
+
 
 def test_json_name_mapping_deserialization() -> None:
     name_mapping = """

--- a/tests/table/test_name_mapping.py
+++ b/tests/table/test_name_mapping.py
@@ -247,11 +247,6 @@ def test_mapping_lookup_by_name(table_name_mapping_nested: NameMapping) -> None:
         table_name_mapping_nested.find("boom")
 
 
-def test_invalid_mapped_field() -> None:
-    with pytest.raises(ValueError):
-        MappedField(field_id=1, names=[])
-
-
 def test_update_mapping_no_updates_or_adds(table_name_mapping_nested: NameMapping) -> None:
     assert update_mapping(table_name_mapping_nested, {}, {}) == table_name_mapping_nested
 

--- a/tests/table/test_name_mapping.py
+++ b/tests/table/test_name_mapping.py
@@ -90,6 +90,7 @@ def test_json_mapped_field_deserialization() -> None:
     """
     assert MappedField(field_id=1, names=["id", "record_id"]) == MappedField.model_validate_json(mapped_field_with_null_fields)
 
+
 def test_json_mapped_field_no_names_deserialization() -> None:
     mapped_field = """{
         "field-id": 1,

